opencti-platform/opencti-front/src/components/ProgressDialog.tsx

```diff
@@ -0,0 +1,165 @@
+import DialogContent from '@mui/material/DialogContent';
+import DialogActions from '@mui/material/DialogActions';
+import Button from '@mui/material/Button';
+import Dialog from '@mui/material/Dialog';
+import React from 'react';
+import LinearProgress from '@mui/material/LinearProgress';
+import Typography from '@mui/material/Typography';
+import DialogTitle from '@mui/material/DialogTitle';
+import makeStyles from '@mui/styles/makeStyles';
+import { useFormatter } from './i18n';
+
+class ProgressDialog {
```

Comment:
Not sure the class is the better approach here, this data should be locally saved in the parent component using the progress bar.

And more of the properties in this class are not progress bar related. The only data used in the progress bar is the percentage to fill. All the rest don't need to be known here.

Reply:
Out of scope. Will be address later.